File: Patika.dev Coderbyte Assessments (Weekly)/Week 4/Java/TwoSum.java
import java.util.*;
import java.io.*;

class Main {

    public static String TwoSum(int[] arr) {
        // code goes here
        int target = arr[0];

        StringBuilder results = new StringBuilder();

        for (int i = 1; i < arr.length; i++){
            for (int j = 1; j < arr.length; j++) {
                if (i != j) {
                    if (arr[i] + arr[j] == target) {
                        if (!(results.toString().contains(arr[j]+","+arr[i]))) {
                            results.append(arr[i] + "," + arr[j] + " ");
                        }
                    }
                }
            }
        }
        return (results.toString().length() > 0) ? results.toString() : "-1";
    }

    public static void main (String[] args) {
        // keep this function call here
        Scanner s = new Scanner(System.in);
        System.out.print(TwoSum(s.nextLine()));
    }

}

File: Patika.dev Coderbyte Assessments (Weekly)/Week 4/Python/Two Sum.py
from itertools import combinations

def all_subsets(arr):
    return list(combinations(arr[1:],2))

def TwoSum(arr):
    # code goes here
    try:
        possible_combinations = list(all_subsets(arr))
        valid_pair_list = []
        string_pair_list = []

        for pair in possible_combinations:
            if sum(pair) == arr[0]:
                valid_pair_list.append(pair)

        if len(valid_pair_list) != 0:
            for pair in valid_pair_list:
                string_pair_list.append(",".join(str(number) for number in pair))

            return " ".join(string_pair for string_pair in string_pair_list)

        else:
            return -1

    except(AttributeError, TypeError):
        return -2

# keep this function call here
print(TwoSum(input()))